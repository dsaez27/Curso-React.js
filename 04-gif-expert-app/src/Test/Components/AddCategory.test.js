import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../Components/AddCategory';

describe('Prueba en <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

	
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

	test('debe de mostrar correctamente <AddCategory />', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de cambiar la caja de textos', () => {
		const input = wrapper.find('input');
		const value = 'hola mundo';
		input.simulate('change', { target: { value } });
		expect(wrapper.find('p').text().trim()).toBe(value);
	});

	test('No debe de postear la informaciÃ³n con submir', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'hola mundo';

        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('input').simulate('submit', { preventDefault() { } });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});
