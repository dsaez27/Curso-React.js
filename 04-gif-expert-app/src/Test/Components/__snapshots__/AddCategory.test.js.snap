// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <AddCategory /> debe de mostrar correctamente <AddCategory /> 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
       
       
    </p>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
